### ISSUE TITLE
2019/7/16 - 2019/12/12

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#198]
1.----------------------------------------------------------------------------------------------------------------------
2.There is no need to be worried if the result is not satisfying. We can adjust parameters, modify algorithms and add new features. (2019-7-16)
3.Search about inference engine; done; (2019-8-22)
4.CS的compound的另一个比喻是计算机技术中的对象, 利用变量和一组对它进行操作的过程表示某个对象; from os p26; this is not good; (2019-8-22)
5.Learn how to cite other papers; list all the result we see; done; (2019-8-22)
6.Since no large-scale dataset of sentential paraphrases exists publicly, we follow Wieting et al. (2017) and automatically generate millions of paraphrase pairs using neural backtranslation.
7.Mallinson et al. (2017) show that neural machine translation (NMT) systems outperform phrasebased MT on several paraphrase evaluation metrics.
8.In this paper, we use the PARANMT-50M corpus from Wieting and Gimpel (2017).
9.The pretrained Czech-English model used for translation came from the Nematus NMT system (Sennrich et al., 2017).
10.----------------------------------------------------------------------------------------------------------------------
11.Specifically, we parse the backtranslated paraphrases using the Stanford parser (Manning et al.,2014),4 which yields a pair of constituency parses hp1, p2i for each sentence pair hs1, s2i, where s1 is the reference English sentence in the CzEng corpus and s2 is its backtranslated counterpart. 
12.Given a sentential paraphrase pair hs1, s2i and a corresponding target syntax tree p2 for s2, we encode s1 using a bidirectional LSTM (Hochreiter and Schmidhuber, 1997), and our decoder is a two-layer LSTM augmented with soft attention over the encoded states (Bahdanau et al., 2014) as well as a copy mechanism (See et al., 2017).
13.Following existing work in NMT (Sennrich et al.,2015), we preprocess s1 and s2 into subword units using byte pair encoding, and we perform decoding using beam search.
14.Use sort similar to optimize show all sentence; done; (2019-8-22)
15.Separate calculate similarity; done; (2019-8-22)
16.Import library属于外码; I know; (2019-8-22)
17.从tensorflow中可以获得启示，我们应该将矩阵乘法从element乘法中分离出去; add matrix multiply to cs; done; (2019-8-23)
18.Download Li Ming’s conference papers; 
19.----------------------------------------------------------------------------------------------------------------------
20.Check software of storing papers; try to use endnote; download mendeley; install; register; done; (2019-8-23)
21.Learn what is LSTM; this is mentioned so many times; 
22.长短期记忆（Long short-term memory, LSTM）是一种特殊的RNN，主要是为了解决长序列训练过程中的梯度消失和梯度爆炸问题。简单来说，就是相比普通的RNN，LSTM能够在更长的序列中有更好的表现。
23.启发式算法（heuristicalgorithm)是相对于最优化算法提出的。一个问题的最优算法求得该问题每个实例的最优解。启发式算法可以这样定义：一个基于直观或经验构造的算法，在可接受的花费（指计算时间和空间）下给出待解决组合优化问题每一个实例的一个可行解，该可行解与最优解的偏离程度一般不能被预计。(2019-8-25)
24.代码审查（英语：Code review）是指对计算机源代码系统化地审查，常用软件同行评审的方式进行，其目的是在找出及修正在软件开发初期未发现的错误，提升软件质量及开发者的技术。代码审查常以不同的形式进行，例如结对编程、非正式的看过整个代码，或是正式的软件检查。(2019-8-24)
25.Modify the structure of our paper; see other papers’ structure; add github address to our paper; done; (2019-8-24)
26.It is experiments, not experiment; add it to our paper; done; (2019-8-24)
27.Place PBS community in the Appendix; done; use A, B, C to write appendix; done; (2019-8-24)
28.Adversarial Examples for Evaluating Reading Comprehension Systems; 1 introduction; 2 The SQuAD Task and Models; 3 Adversarial Evaluation; 4 Experiments; 5 Discussion and Related Work;
29.----------------------------------------------------------------------------------------------------------------------
30.Adversarial Over-Sensitivity and Over-Stability Strategies for Dialogue Models, 1 introduction; 2 tasks and models; 3 Adversarial Strategies; 4 Adversarial Training; 5 Experimental Setup; 6 Result; 7 Byte-Pair-Encoding VHRED; 8 Related Works; 9 Conclusion; Acknowledgments
31.AdvEntuRe: Adversarial Training for Textual Entailment with Knowledge-Guided Examples, 1 Introduction, 2 Related Work, 3 Adversarial Example Generation, 4 AdvEntuRe, 5 Experiments, 6 Conclusion; 
32.Breaking NLI Systems with Sentences that Require Simple Lexical Inferences; 1 Introduction; 2 Background; 3 Data Collection; 4 Evaluation; 5 Analysis; 6 Conclusion; 
33.Generating Natural Language Adversarial Examples; 1 Introduction; 2 Natural Language Adversarial Examples; 3 Attack Design; 4 Experiments; 5 Conclusion; Acknowledgement; 
34. As we can see, paper uploaded are written by Latex; 
35.在归纳论文段落大意的时候，要尽量避免用作者临时发明的术语，尽量用平实易懂的语言。Done; (2019-8-25)
36.I think reading Ming Li’s work is very import for us; how to extract from github is also useful; since it is too hard to classify all codes on our own; (2019-8-25)
37.Points: 1, how to process source code; 2, how to get plenty of code from github; (2019-8-25)
38.Pay attention the paper they referred to; 
39.启发式算法体现了道法自然的思想；done; (2019-8-25)
40.----------------------------------------------------------------------------------------------------------------------
41.Download adobe reader; done; not good as I think; (2019-8-25)
42.Learning Unified Features from Natural and Programming Languages for Locating Buggy Source Code; find bug; 
43.Enhancing the unified features to locate buggy files by exploiting the sequential nature of source code; find bug; 
44.Distributed representations of words and phrases and their compositionality
45.Convolutional neural networks for sentence classification
46.Sequence to sequence learning with neural networks
47.As we can see, the paper has many previous papers; pay attention to the previous papers; the paper’s previous paper are: Huo, Li, and Zhou 2016; Huo and Li 2017; (2019-8-25)
48.Let us pay attention to Huo and Li 2017; 
49.Modify introduction to illustrate the reason why our method is important; done; modify abstract; done; (2019-8-26)
50.----------------------------------------------------------------------------------------------------------------------
51.Modify introduction; done; (2019-8-26)
52.Try to download X-mind; install to learn to use; Learning Unified Features from Natural and Programming Languages for Locating Buggy Source Code; done; (2019-8-26)
53.Disable protected mode of adobe reader; done; (2019-8-26)
54.Use measure to replace calculate in our paper; done; (2019-8-26)
55.The Latex result is not as good as I think; so before get template, continue using makedown format; (2019-8-26)
56.Continue reading the paper Enhancing the Unified Features to Locate Buggy Files by Exploiting the Sequential Nature of Source Code; 
57.虽然一篇文章可能有很多引用的文献，但它们的重要程度是不同的。那些被反复提起的文献是比较重要的，那些只提一遍的很可能只是为了表达某个知识的来源出处(甚至是为了凑字数)。我们只需要关注那些被反复提起的引述文献就可以了。(2019-8-26)
58.Confirm that this year is the 37th conference; done; (2019-8-27)
59.The chapters have already been checked and approved by the contributing authors so we can no longer make corrections to the chapters; done; (2019-8-27)
60.----------------------------------------------------------------------------------------------------------------------
61.To make sure that no errors have been inadvertently introduced into the cover, front matter and author index, we would ask you to kindly check the paper titles and author names both in the finalized table of contents and in the author index; 
62.Deal with file sent by He; done; (2019-8-27)
63.It seems that the result of SC is not stable; let us try our method on another machine; put necessary files in the U-disk; done; (2019-8-27)
64.Download free download manager; give up; download xunlei; done; start download ubuntu 16; Not successful; strange, only 14 can be installed; (2019-8-28)
65.Let us create a new account on the computer; on the desktop the result is the same; let us try in virtual box; 
66.Continue reading paper; check other paper how they represent the number of elements; the answer is clear, n1 and n2; (2019-8-28)
67.Read paper of Zhou 2012; replace current to query; done; move 5.2 add and edit sentence to appendix B4; done; (2019-8-28)
68.Replace compound with sentence; use knowledge base to replace database; done; (2019-8-28)
69.代码分析比机器翻译有用得多；哪家公司不用写代码？又有几家公司做机器翻译呢？(2019-8-28)
70.----------------------------------------------------------------------------------------------------------------------
71.Find a way to compile all sentences and update; we should put this function in the analyzer; (2019-8-28) 
72.Github: location already exists; solve the problem; (2019-8-28)
73.Change function sequence in the analyzer; add number to each function; (2019-8-28)
74.add show all sentences in the analyzer; try to show all sentences in the analyzer; done; (2019-8-28)
75.Error; Notice: Undefined index: in C:\wamp64\www\similarity-comparison\find_common.php on line 233; give up; (2019-8-28)
76.How can we compile one sentence in the analyzer; hold;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
77.Search zhihu for what is Markov; example of wangergou; done; (2019-8-29)
78.Search zhihu for Monte Carlo method; done; (2019-8-29)
79.Replace sort with rank; done; add -ing to verb; done; (2019-8-29)
80.----------------------------------------------------------------------------------------------------------------------
81.Sometimes it is difficult to locate bugs in our system; find a way to locate position of bugs; 
82.Try to show lexical error in our program; success; (2019-8-29)
83.Try to show syntax error in our compiler; try to use $SymbolSequence to test the result; yes, correct; make generate_id_sequence to show syntax error; done; (2019-8-29)
84.Deal with the problem that == has syntax error; != is successful; now == is successful, too; done; (2019-8-29)
85.delete process_conjunction; done; deal with < ; done; deal with >; done; deal with ~; done; (2019-8-29)
86.CS 类似于人工智能中的专家系统; 给theory中的(1) Identify the task. (2) Assemble the relevant knowledge, 结合CS具体问题具体分析，解释每一个步骤的意思; (2019-8-22)
87.Check wiki for vector space model; 
88.Each dimension corresponds to a separate term. If a term occurs in the document, its value in the vector is non-zero. Several different ways of computing these values, also known as (term) weights, have been developed. One of the best known schemes is tf-idf weighting (see the example below).
89.Typically terms are single words, keywords, or longer phrases. If words are chosen to be the terms, the dimensionality of the vector is the number of words in the vocabulary (the number of distinct words occurring in the corpus).
90.----------------------------------------------------------------------------------------------------------------------
91.Relevance rankings of documents in a keyword search can be calculated, using the assumptions of document similarities theory, by comparing the deviation of angles between each document vector and the original query vector where the query is represented as a vector with same dimension as the vectors that represent the other documents.
92.How to write xieti in markdown? *d*, *q*; change Picture 1 and 2 to 2 and 3; it seem change filename directly on github; give up; (2019-8-30)
93.Register new semester in HUST; (2019-8-30)
94.Read wiki of tf-idf; 
95.tf-idf（英语：term frequency–inverse document frequency）是一种用于信息检索与文本挖掘的常用加权技术。tf-idf是一种统计方法，用以评估一字词对于一个文件集或一个语料库中的其中一份文件的重要程度。字词的重要性随着它在文件中出现的次数成正比增加，但同时会随着它在语料库中出现的频率成反比下降。
96.在一份给定的文件里，词频（term frequency，tf）指的是某一个给定的词语在该文件中出现的频率。这个数字是对词数（term count）的归一化，以防止它偏向长的文件。
97.周志华: Forest autoencoder; deep forest; 不清楚深度森林前景如何; (2019-8-30)
98.词袋模型: 一篇文档里面有很多很多句子，每个句子又是由一个个的词组成。词袋模型，通俗地讲，就是：把一篇文档看成词袋，里面装着一个个的词。从而，将一篇文档转化成了一个个的词(或者称之为 term)，显然地，文档转化成一个个的词之后，词与词之间的顺序关系丢失了。(2019-8-30)
99.We should put the object code result in the table; add position of object code and object code symbol in print_sentence.php; done; (2019-8-30)
100.----------------------------------------------------------------------------------------------------------------------
101.How can we define html element in php environment? Give up; add object code in the analyzer; done; learn how to do column span; done; (2019-8-30)
102.Translate object code in object symbol in analyzer; hold;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
103.Should we consider the tf-idf in our program? Replace calculate with compute; done; (2019-8-31)
104.有没有可能，因为我们不断向数据库中添加数据，similarity的数值才会变化？(2019-8-31)
105.TRIZ中的物场分析其实就是数学中的集合和运算, 也类似于word的词性; (2019-8-31)
106.Add intuition of similarity to theory; give up, since we already have space vector model; (2019-8-31)
107.Replace calculate with compute; done; change sort similarity to rank similarity; change button name; done; change sort to rank; done; (2019-8-31)
108.separate confidence out as a parameter; done; (2019-8-31)
109.Register for dormitory; done; (2019-8-31)
110.----------------------------------------------------------------------------------------------------------------------
111.Check whether we have 5 computer classes; hold;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
112.Compute the reverse similarity; hold;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
113.校历; done; 学籍异动流程; done; 中期筛选; done; 申请博士学位论文规定; done; 学术硕士培养方案; done; 工程硕士培养方案; done; 名单; done; 课程表; done; (2019-8-31)
114.Replace calculate with compute in find_common.php; done; add confidence in compute_similarity; done; separate compute_similarity; done; replace it find_common.php; done; replace it in rank_similarity.php; done; (2019-8-31)
115.Add inverse calculate similarity, the result is not as good as I think; 
116.What is CVS; 没有光明的路咱都走下来了，更何况现在已经有了一丝光明？
117.How to print dot in markdown format? Give up; get package of hualushui; (2019-9-1)
118.Modify generate_id_sequence; add symbol_to_id; done; delete array_category in generate_id_sequence; done; keep array_category in preprocess; (2019-9-1)
119.Try to show object code symbol in print_sentence.php; in find common to see how they split the words; change object_code to object_code_list; done; 
120.----------------------------------------------------------------------------------------------------------------------
121.机器翻译很吃硬件, 很吃内存, 需要很多卡, 而这边没有足够的硬件; 光跑强化就跑不起来; 关键要做; 多做对话, 聊天机器人, 发论文导向; 在看一个综述，90多页; 在机器上跑了半天, 进展缓慢; (2019-9-1)
122.Change $para to $sentence_id; done; check source code for their format; <br>; (2019-9-1)
123.Add sentence editor compiler; done; modify translate_source.php to make it to become a function;;;;;;;; move position of sql query; done; delete the database sentence; done; (2019-9-2)
124.Get gloves; done; download fujian from master1903; already have; done; figure out courses that belong to computer science; done; (2019-9-2)
125.Write progress report to Prof. He; done; change show select sentence to show selected sentence; done; change show_select_sentence.php to show_selected_sentence.php; done; (2019-9-2)
126.Separate first line and second line; done; (2019-9-2)
127.Notice: Undefined index: ImportLibrary in C:\wamp64\www\similarity-comparison\translate_source.php on line 79; deal with ImportLibrary in translate source; add $ImportLibrary parameter in function preprocess; done; move $_POST['ImportLibrary'] to translate_source; done; (2019-9-2)
128.Add button Translate All Sentences; done; append show all sentences function to it; done; (2019-9-2)
129.Where do they find the Eclipse files and bug reports? We should try to find it in experimental setup; 
130.----------------------------------------------------------------------------------------------------------------------
131.https://bugs.eclipse.org/bugs/show_bug.cgi?format=multiple&id=80720
132.http://www.eclipse.org
133.http://www.eclipse.org/swt/
134.CVS（英语：Concurrent同时 Versions System，或Concurrent Versioning System）代表协作版本系统或者并发版本系统，是一种版本控制系统，方便软件的开发和用户协同工作。(2019-9-2)
135.Apache软件基金会（Apache Software Foundation，简称为ASF），是专门为支持开源软件项目而办的一个非营利性组织。在它所支持的Apache项目与子项目中，所发行的软件产品都遵循Apache许可证（Apache License）。(2019-9-2)
136.统一的版本号。CVS是对每个文件顺序编排版本号，在某一时间各文件的版本号各不相同。而Subversion下，任何一次提交都会对所有文件增加到同一个新版本号，即使是提交并不涉及的文件。所以，各文件在某任意时间的版本号是相同的。版本号相同的文件构成软件的一个版本。(2019-9-2)
137.原子提交。一次提交不管是单个还是多个文件，都是作为一个整体提交的。在这当中发生的意外例如传输中断，不会引起数据库的不完整和数据损坏。(2019-9-2)
138.重命名、复制、删除文件等动作都保存在版本历史记录当中。(2019-9-2)
139.Register as new student; done; submit hukou migrate; done; get body examine table; done; (2019-9-3)
140.----------------------------------------------------------------------------------------------------------------------
141.Fill in the QQ table online; done; transfer 7 yuan to class president; done; (2019-9-3)
142.整理dropbox中不需要的文件；(2019-3-23)
143.Attend opening ceremony; done; (2019-9-4)
144.First after reading the whole paper; as we can see, the previous are concept location, and this work is bug localization; so this is the edge; (2019-9-4)
145.Try to find bug database and source file; 
146.All projects have complete bug database and change history, and have different numbers of bugs and source code files; (2019-9-5)
147.Both Eclipse and AspectJ use the Bugzilla bug tracking system and the CVS/SVN version control system; (2019-9-5)
148.For each subject system, we collect its initial bug reports from the bug tracking system (such as BugZilla). To evaluate the bug localization performance, we only collect the bug reports of fixed bugs. (2019-9-5)
149.Check BugZilla; download BugZilla; 
150.----------------------------------------------------------------------------------------------------------------------
151.矩阵论; 现代计算机网络; 并行处理; 现代数据管理理论与技术; (2019-9-5)
152.人工智能; 计算的基本理论; (2019-9-5)
153.计算智能; 模式识别; (2019-9-5)
154.What is Bugzilla?
155.Bugzilla is a robust, featureful and mature defect-tracking system, or bug-tracking system. Defect-tracking systems allow teams of developers to keep track of outstanding bugs, problems, issues, enhancement and other change requests in their products effectively. Simple defect-tracking capabilities are often built into integrated source code management environments such as Github or other web-based or locally-installed equivalents. We find organizations turning to Bugzilla when they outgrow the capabilities of those systems - for example, because they want workflow management, or bug visibility control (security), or custom fields. (2019-9-5)
156.Bugzilla is both free as in freedom and free as in price. Most commercial defect-tracking software vendors charge enormous licensing fees. Despite being free, Bugzilla has many features which are lacking in both its expensive and its free counterparts. Consequently, Bugzilla is used by hundreds or thousands of organizations across the globe. (2019-9-5)
157.Bugzilla is a web-based system but needs to be installed on your server for you to use it. However, installation is not complex. (2019-9-5)
158.Get moon cakes; done; bring it to the lab; (2019-9-5)
159.Try to buy tape online; give up; (2019-9-5)
160.----------------------------------------------------------------------------------------------------------------------
161.SWT bug reports are stored in the Eclipse Bugzilla database; when we search google for Eclipse, we have https://bugs.eclipse.org/bugs/; register an Eclipse account; done; (2019-9-5)
162.Set user preferences; general preferences; done; email preferences; done; saved searches; done; account information; done; API keys; done; permissions; done; (2019-9-5)
163.湖北 武汉市 洪山区 珞喻路华中科技大学东校区创新研究院907室; done; (2019-9-5)
164.Search; simple search; STATUS, open, closed, all; done; PRODUCT; ECD; loT; location Tech; Eclipse; EE4J; Eclipses foundation; modeling; Mylyn; RT; SOA; technology; tools; web tools; science; this is classification; (2019-9-6)
165.我们似乎一脚踏进了软件工程的大门？(2019-9-6)
166.Advanced search; 
167.Status, The Status field indicates the current state of a bug. Only certain status transitions are allowed; resolution, The Resolution field indicates what happened to this bug; (2019-9-6)
168.Open bugs; 
169.Unconfirmed; This bug has recently been added to the database. Nobody has confirmed that this bug is valid. Users who have the "canconfirm" permission set may confirm this bug, changing its state to CONFIRMED. Or, it may be directly resolved and marked RESOLVED. (2019-9-6)
170.----------------------------------------------------------------------------------------------------------------------
171.CONFIRMED; This bug is valid and has recently been filed. Bugs in this state become IN_PROGRESS when somebody is working on them, or become resolved and marked RESOLVED. (2019-9-6)
172.IN_PROGRESS; This bug is not yet resolved, but is assigned to the proper person who is working on the bug. From here, bugs can be given to another person and become CONFIRMED, or resolved and become RESOLVED. (2019-9-6)
173.Eclipse is in classification; 
174.No resolution yet. All bugs which are in one of these "open" states have no resolution set. (2019-9-6)
175.Closed Bugs
176.RESOLVED: A resolution has been performed, and it is awaiting verification by QA. From here bugs are either reopened and given some open status, or are verified by QA and marked VERIFIED. (2019-9-6)
177.VERIFIED: QA has looked at the bug and the resolution and agrees that the appropriate resolution has been taken. This is the final status for bugs. (2019-9-6)
178.FIXED: A fix for this bug is checked into the tree and tested. (2019-9-6)
179.INVALID: The problem described is not a bug. (2019-9-6)
180.----------------------------------------------------------------------------------------------------------------------
181.WONTFIX: The problem described is a bug which will never be fixed. (2019-9-6)
182.DUPLICATE: The problem is a duplicate of an existing bug. When a bug is marked as a DUPLICATE, you will see which bug it is a duplicate of, next to the resolution. (2019-9-6)
183.WORKSFORME: All attempts at reproducing this bug were futile, and reading the code produces no clues as to why the described behavior would occur. If more information appears later, the bug can be reopened. (2019-9-6)
184.Order air conditioner controller online; done; clean table in dormitory; done; (2019-9-6)
185.在阅读一篇论文被某个问题卡壳时, 转而阅读它引述的文献或引述它的文献, 因为在一篇论文中一笔带过的问题很可能在另一篇论文里有详细解答; (2019-9-6)
186.Combining Deep Learning with Information Retrieval to Localize Buggy Files for Bug Reports
187.Which product do they use? JDT; Huo said which product they use; check Lam 2015’s experiment; find dataset at https://figshare.com/articles/The_dataset_of_six_open_source_Java_projects/951967; great; (2019-9-6)
188.Map bug report with fixed files; what is heuristics proposed by Dallmeier and Zimmermann? Hold; (2019-9-6)
189.Investigate file 951967; ReadMe; 
190.----------------------------------------------------------------------------------------------------------------------
191.Introduction; dataset to evaluate rank approach; (2019-9-7)
192.How to obtain the dataset; excel format; XML format; database file SQL; (2019-9-7)
193.Dataset structure; The bug reports and commit history of AspectJ; The bug reports and commit history of Birt; The bug reports and commit history of Eclipse Platform UI; The bug reports and commit history of JDT; (2019-9-7)
194.What is XML? How to read them? 
195.硕士研究生校级公共课《第一外国语（英语）》免修：英语统考70分以上（不含单独考试的考生）和免试推荐硕士研究生以及直博生可直接免修，不用提交申请。研究生院会直接录入“免修”成绩。研究生在制订个人培养计划中需将课程列入计划，学期一律选“第一学期”。(2019-9-8)
196.Select English; done; select 一级学科基础课; done; change time of doing research; done; (2019-9-9)
197.Give fapiao of paper to Xuelian Liu; wait; ;;;;;;;;;;;;;;;;;;;;;;
198.Check content of each course; 多媒体基础; done; 现代数据管理理论与技术; done; 现代计算机网络; done;;;; 人工智能; done; 计算的基本理论; done;;;;;; 计算智能; done; 模式识别; done; (2019-9-10)
199.Buy notebook online; done; download textbook of 矩阵论; done; (2019-9-10)
200.----------------------------------------------------------------------------------------------------------------------
201.Examine course list and locations; done; (2019-9-5)
202.Learning to Rank Relevant Files for Bug Reports using Domain Knowledge; I think a better way is to read Xin Ye’s papers; (2019-9-10) 
203.Compare downloaded bug report and what is on the web; read it again to figure out what they are doing; we are trying to reproduce Jian Zhou’s experiment; we are trying to figure out how they set up the experiments; 
204.They use the product of Eclipse platform; done; (2019-9-10)
205.The website is https://bugs.eclipse.org/bugs/show_bug.cgi?format=multiple&id=80720 (2019-9-10)
206.Summary, Pinned console does not remain on top; (2019-9-10)
207.Description, * Open two console views * Turn off the option to remove terminated launches * Launch one program that produces output * Pin one console * Launch another program that produces output * Both consoles display the last launch > The pinned console should remain pinned. (2019-9-10)
208.we also find summary and description in data set provided by Xin Ye; (2019-9-10)
209.Where do they find source code files? I think we should install Eclipse to understand better; install java first; done; (2019-9-10)
210.----------------------------------------------------------------------------------------------------------------------
211.JDK:Java Development Kit(Java开发工具包); JRE:Java Runtime Environment(Java运行时环境); JVM:Java Virtual Machine(Java虚拟机); (2019-9-10)
212.JDK=JRE+Java常用的类库+Java工具; JRE=JVM+java常用的类库; (2019-9-10)
213.Download java JDK; register oracle account; done; download Eclipse; install Eclipse; done; write hello world in java; (2019-9-10)
214.JavaSE（J2SE）（Java2 Platform Standard Edition，java平台标准版）; JavaEE(J2EE)(Java 2 Platform,Enterprise Edition，java平台企业版); JavaME(J2ME)(Java 2 Platform Micro Edition，java平台微型版); (2019-9-10)
215.Write weekly report; done; (2019-9-10)
216.Buy text book of matrix; expensive? Ask about book store; 西学书店/绝望坡学友书店; ask Jianhong to check the room; get money back; done; return the key to Jianhong Li; ;;;;;;; (2019-9-11)
217.Where do they find source code of Eclipse? How do they know the relevant source code files? 
218.Read project; BugLocator ranks all files based on the textual similarity between the initial bug report and the source code using a new similarity measure, taking into consideration information about similar bugs that have been fixed before; ReLink is the tool we developed for our research work on recovering missing links between fixed bugs and committed changes. (2019-9-11)
219.files -- contains the full path of every Java file that was fixed in this commit; (2019-9-11)
220.----------------------------------------------------------------------------------------------------------------------
221.what is default directory of git clone in windows? It’s in user; (2019-9-11)
222.Try to download source code of Eclipse;  git clone https://git.eclipse.org/r/p/platform/eclipse.platform.ui ; strange, download faied, failed again;  
223.Read how to contribute webpage of Eclipse; Report Bugs and Enhancements; Fix Bugs or Implement Enhancements; find source code of Eclipse; https://git.eclipse.org/c/; its name is Eclipse Git repositories; (2019-9-11)
224.I find “Recovering Links between Bugs and Changes”, is this paper mentioned in “Where Should the Bugs Be Fixed”? no, strange; so what is the relationship between ReLink(2011) and BugLocator(2012)? Read the paper of relink, since another paper also mentioned commit changes; No, “where should the bugs be fixed” does not use the method in “Recovering Links between Bugs and Changes”, they use the traditional heuristics proposed by Bachmann and Bernstein [5]; its name is “Software Process Data Quality and Characteristics-A Historical View on Open and Closed Source Projects”; (2019-9-12)
225.Read wiki of ReLink; introduction; This page describes the tool we developed for our research work on recovering missing links between fixed bugs and committed changes, utilizing Information Retrieval technology and other useful features; (2019-9-12)
226.So what is the source code they use in “Where Should the Bugs Be Fixed”? Previous Source code, next source code, or commit changes? It is the whole source file, they “performs lexical analysis for each source code file and creates a vector of lexical tokens” as mentioned in the paper; so is the source code previous version, or next version? They do not mention it, but we can infer that they are current code, since there is only one version of source as we downloaded; (2019-9-12) 
227.let us download the Eclipse source code file; in the name line, we can see categories of each product; strange, failed again; download source code of BugLocator; done; great, we find bug data in BugLocator source code; another file bug locator.zip is deprecated; (2019-9-13)
228.Try to set environmental variables for java; find path of java where it is installed; done; javac and java -version can be used; done; (2019-9-13)
229.Try to run the program in “Where Should the Bugs Be Fixed”, following their instruction; read wiki of BugLocator; After unpacking the zip file, you can find the following files: * BugLocator.jar * ReadMe.txt * Data Folder(xml examples); -b indicates the bug information file,the bug information file must be an .xml file in the data folder; -s indicates the source code directory; -a indicates the alpha factor for combining VSMScore and SimiScore; -o indicates the result file,the format of result file is "bug id, relevant source code file,rank(start with 0),score"; (2019-9-13)
230.----------------------------------------------------------------------------------------------------------------------
231.B: bug information file xml; C:\Users\take_\Downloads\BugLocator\data\SWTBugRepository.xml
232.S: indicates the source code directory; C:\Users\take_\Downloads\swt-3.1\src
233.A: indicates the alpha factor for combining VSMScore and SimiScore; 0.2
234.O: indicates the result file; output.txt
235.java -jar BugLocator.jar -b C:\Users\take_\OneDrive\liming\where\BugLocator\data\SWTBugRepository.xml -s C:\Users\take_\OneDrive\liming\where\swt-3.1\src -a 0.2 -o output.txt
236.Error: unable to access the jarfile BugLocator; search google for solutions; add .jar to execute file; success; we get the result; (2019-9-13)
237.I think we should create an independent folder for “Where Should the Bugs Be Fixed”; copy downloaded file into this new created file; done; try to run the java program again; finally, their program does not allow long filename; (2019-9-13)
238.ReadMe of BugLocator; it is the same as written on the website; (2019-9-13)
239.Try to investigate their generated result; format is  "bug id, relevant source code file, rank(start with 0), score"; bug id is number; relevant source code file is file name; rank is number, score is number; (2019-9-13)
240.----------------------------------------------------------------------------------------------------------------------
241.The steps are: (1) create bug corpus; (2) create bug vector; (3) compute bug similarity; (4) create code corpus; (5) compute SimiScore; (6) create index; (7) create vector; (8) compute VSM score; (9) compute LengthScore; (10) evaluate; 
242.Try to find source code of “Where Should the Bugs Be Fixed”; download bug locator.zip and check what is inside; it is the same but already deprecated; (2019-9-13)
243.How do I ask the authors of a scientific research paper to share the data and source code they used? Jian Zhou; zhoujian1286@yahoo.com.cn; find mail system of HUST; search Jian Zhou’s information; register ResearchGate; they refuse HUST email; (2019-9-14)
244.由于科研任务的复杂性和繁重性, 使用正面进攻的策略往往收效甚微而且影响心情; 更好的策略是碉堡式推进, 一点点扫清外围的知识点, 逐渐收缩包围圈, 最终抵达胜利; 这样做的好处是有效减轻心理负担, 并且增加成就感; (2019-9-11)
245.在做一件事情之前, 先分析正确性和可行性, 是数学思维的体现; (2019-9-14)
246.So before we start, let’s evaluate the result generated by the program; format is  "bug id, relevant source code file, rank(start with 0), score"; I think we should store the result in a xml file; 
247.What is more strange is that they use java as programming language; what is the most common used language in machine learning? Anaconda已经自动为你安装了Jupter Notebook及其他工具，还有python中超过180个科学包及其依赖项; download anaconda; done; install anaconda; done; (2019-9-14)
248.Try to read this output file with python; #with语句自动调用close()方法; In this case, every element in the list contain a \n in the end the string, which would be extremely annoying in some cases; python read text file into two dimensional array; done; with open("output.txt") as textFile: lines = [line.rstrip('\n').split(",") for line in textFile]; (2019-9-14)
249.Try to save two dimensional array with python into xlsx; give up; try to solve two dimensional array to csv file; give up; print two dimensional array to the screen; give up; done; (2019-9-14)
250.----------------------------------------------------------------------------------------------------------------------
251.算法的五个基本特性; 确定; 可行; 有穷; (2019-9-14)
252.Send email to the author to ask for source code; the system sent my email back; (2019-9-14)
253.Try to find 果然有味 and 学友书店; we find the fruit store on despair hill; 
254.Search the same bug in another data set to check whether relevant file in this paper is correct; we cannot find them; let us assume they are correct; evaluate this result using their own metrics; what is their metrics? 
255.Let us try another program we find on github; download the python program; 无论如何, 我们要亲自编代码, 因为其中的某些思想, 和后面的几篇文章一脉相承; 而且便于我们发现自己不懂的问题; 而且对建立我们的信息意义重大; (2019-9-14)
256.Following the ten step to write our own code; (1) create bug corpus; (2) create bug vector; (3) compute bug similarity; (4) create code corpus; (5) compute SimiScore; (6) create index; (7) create vector; (8) compute VSM score; (9) compute LengthScore; (10) evaluate; 
257.Figure out each file in the temp folder; BugCorpus; as we can see, the result number does not equal to bug number, as some bugs are relevant to multiple files; perhaps they are keywords in each bugs; 98 bugs, 265 results; search Eclipse website to check whether they are correct; 14654.txt, correct; 58185.txt, correct; (2019-9-15)
258.BugSimilarity.txt, does not know what is it; BugTermList.txt, does not know; BugVector.txt, hmm; ClassName.txt, ???; CodeCorpus, ???; CodeVector, ???; FixLink.txt, ??? IDC, ??? LengthScore.txt, seem understand; SimiScore, seem understand; ????? (2019-9-15)
259.Check whether fix link show in the result; yes; done; (2019-9-15)
260.----------------------------------------------------------------------------------------------------------------------
261.Figure out how to create bug vector; this belong to the process of indexing; as we can see, the number of text in corpus and in bug vector are different; let’s check whether the vector of each file is same; No, they are different; Easy to understand, they are weights; so 0, 1, 2 might be word index; check whether they have word list, yes they have; (2019-9-15)
262.Check whether word in word list corresponds to bug vector and bug corpus; now we need to count number of kinds of words in bug locator; we need to index words in a text; strange, WordList showing in desktop is different from what is showing on the website! WHAT?; okay, we can trust WordList now; (2019-9-15)
263.WordList contain word both in bug corpus and code corpus; we test 6672 and 6671, and there is no 6673; (2019-9-15)
264.So what is the meaning of the number before “:” in the file BugVector? It is the index, however, the index in the WordList is different from what in the Bug index; let us keep the word which are in the word list for every bug corpus; do this for 100040.txt; (2019-9-15)
265.Check how to create numpy array; WordList = np.zeros((len(lines), 3)); done; assign value from text to WordList; how to get shape of numpy array? Numpy array are all numbers, so it should be used here; (2019-9-16)
266.Buy pens online; download PPT of distributed system; (2019-9-16)
267.Generate word list; store word list into an array; done; does python has map function? Yes, but map in python is different with map in C++; (2019-9-16)
268.add a third column to WordList named frequency; it is append; done; change variable name lines to WordList; done; (2019-9-16)
269.If a word is not in the word list, delete it; if it has already appeared in the article, delete it and add frequency; try to read “100040.txt” into list; print it in local, can we set certain path? "C:\\Users\\take_\\OneDrive\\liming\\where\\BugLocator\\tmp\\BugCorpus\\100040.txt; done; (2019-9-16)
270.----------------------------------------------------------------------------------------------------------------------
271.Register for bath water; (2019-9-17)
272.Change 100040.txt to 100095.txt; done; change back; keep words only in the word list; done; delete word that has already occurred once; done; (2019-9-17)
273.Figure out the map between word list and bug vector the author produce; read bug vector and delete the weight; read bug vector; done; (2019-9-17)
274.Separate as a function; how to define function in python? Done; strange, the number of words generate in 100095.txt is 23, but the answer is 25? Is it possible that the index are generated from term list? How to change one dimension list to two dimension in python? Done; WHAT? BugTermList is still wrong; as we can see from the Internet, the summary and description are both contained in the corpus; (2019-9-17)
275.Read all files into a list; done; modify generate_BugCorpus; done; (2019-9-18)
276.(unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape; This error occurs because you are using a normal string as a path; done; (2019-9-18)
277.We can use the index bug vector directly, and check whether we can get the weight; the problem is, if we use number directly, we cannot count the frequency; 
278.Let us take “due” in document 100040.txt for example; read 100040.txt; done; count “due” in document 100040.txt; strange, their result is much smaller than 1; finally, we have to do it our own; (2019-9-19)
279.Now we have bug vector and code vector, how to compute the similarity? It is illustrated in Formula (7); is the #docs for bug same as #docs for program? Perhaps they combine all document as a whole? (2019-9-19)
280.----------------------------------------------------------------------------------------------------------------------
281.Let us check which files are contained in; as we can see, all files are java files; Lucky, we have code corpus; deal with code corpus; save all code words into a list; separate word vector by certain words; replace \n and \t; python split multiple separators; we just use .java and it is OK now; (2019-9-19)
282.Calculate weight for each word in 100040.txt; we should do this later; separate CodeList as each single program file; change files to BugFiles; done; python get the last 5 letter in a string; done; python delete first element; done; now we get CodeFiles and BugFiles; (2019-9-20)
283.Calculate ftd; ftd is the same; done; calculate #docs; #docs is the total number of bug files and code files; change generate_BugCorpus to generate_OneCorpus; done; I think we should combine all bug corpus together; python list add element at first; done; (2019-9-20)
284.check what we generate for code file; add function generate_OneCodeCorpus; change BugCorpus to ThisBugCorpus; done; now we generate ThisCodeCorpus; done; generate CodeCorpuses; done; (2019-9-20)
285.Calculate nt in this circumstances; 
286.it can be seen that the two conclusions from the theoretical point of view have some differences. Therefore, empirical analysis on these problems has become very valuable. It is the basis of verification theory and has intuitive persuasiveness. 可以看出，上述二者，从理论分析角度得出的结论存在者一定的差异性。所以，实证角度研究这些问题，就变得非常具有现实意义和价值，它是验证理论的基础，具有直观的说服力。We can see that the two conclusions have some differences theoretically. Thus, it is highly desirable to analyze these problems empirically so as to provide intuitions and proof to the theory. (2019-9-20)
287.Change nt to ntBug; done; count nt in ntCode; done; add them; done; (2019-9-20)
288.Unsubscribe QQ news; (2019-9-21)
289.Check Jupyter Notebook cell; give up; (2019-9-21)
290.----------------------------------------------------------------------------------------------------------------------
291.Compute weight for each word in bug corpus; can python define class? Yes; add class Word; let us check whether the python class works; yes, it works; (2019-9-21)
292.change WordList to WordListTwo; done; create new WordList; done; modify generate_OneBugCorpus, replace WordListTwo with WordList; done; modify generate_OneCodeCorpus, replace WordListTwo with WordList; done; (2019-9-21)
293.Replace WordListTwo with WordlistContent; done; it is obvious that we need to use class in order to find common word for bug corpus and code corpus; move class content to the top; done; if we use class to represent element, how should we deal with the title? Let us use class to solve this problem; (2019-9-21)
294.Modify generate_OneBugCorpus to use class Corpus in it; done; generate new BugCorpuses; done; Modify generate_OneCodeCorpus to use class Corpus in it; done; generate new CodeCorpuses; done; (2019-9-21)
295.Python check whether an element exist in the list; element in list; use in syntax to modify generate_OneBugCorpus; give up, since we also need to count frequency; modify the last segment to adapt to the change; done; (2019-9-21)
296.Use element to replace word in CorpusMembers; use Element in generate_OneBugCorpus; python print class list; give up; done; use Element in generate_OneCodeCorpus; done; (2019-9-21)
297.Read notes of distributed systems; (2019-9-21)
298.Bug corpus output frequency at the same time;  there is another way to do this, after dealing with all words in a document, then create corpus at once; Use this way to modify generate_OneBugCorpus; create dictionary, which is the WordName column of WordList; give up; (2019-9-21)
299.add frequency in class Element; done; output bug corpus with their word frequency; done; modify generate_OneCodeCorpus to create at once; done; print code element frequency; done; change BugFile to ThisBugFile; done; (2019-9-22)
300.----------------------------------------------------------------------------------------------------------------------
301.rewrite the last segment using frequency we calculate; let us use icon for example; get ftd using new way; done; get ntBug using new way; done; get ntCode using new way; done; (2019-9-22)
302.we need to separate calculate weight as a unique function; done; compute weight for each word in 100040.txt; done; compute weight for each word in ALL bug corpus; done; We need to write compute weight for code file; done; (2019-9-22)
303.Download slides of intelligent computing; done; search what is 二分图; 二分图的顶点可以分成两个互斥的独立集 U 和 V ，使得所有边都是连结一个 U 中的点和一个 V 中的点; done; (2019-9-22)
304.匹配; 给定一个二分图 G，在G 的一个子图 M 中，M 的边集中的任意两条边都没有共同的端点，则称 M 是一个匹配; done; 完美匹配; 包含了图G中所有原来顶点的匹配; done; (2019-9-22)
305.顶点覆盖; 图G的顶点覆盖是一个顶点集合V，使得G中的每一条边都接触V中的至少一个顶点; done; |M| is the number of edge; (2019-9-22)
306.Lower bound is necessary; done; under stand the first algorithm for allocating jobs; (2019-9-22)
307.In order to easily understand whether we are right, let us create our own code corpus and bug corpus and word list; create folder tmpMine; done; change CodeCorpus.txt to their own folder; done; (2019-9-24)
308.delete CodeCorpus to only 3 members in tmpMine folder; done; change Wordlist to aaa bbb ccc; done; delete BugCorpus to only 3 members, and change each folder to aaa bbb ccc; done; now we find it is the Wordlist problem; change it path; done; (2019-9-24)
309.Evaluate whether weight is correct or not; now the weight we get is zero; change 3 code files to a a a, b b b, c c c; done; why the weight of C is different with A and B; strange, the weight of C is different; #docs is correct, which means len(CodeCorpuses) is WRONG; add the last ThisCodeFile to CodeFiles; done; (2019-9-24)
310.----------------------------------------------------------------------------------------------------------------------
311.Document vector is made up with element weight; add document norm to class Corpus; done; delete element weight in the initial parameters; done; add function compute_BugCorpus_norm; done; compute Norm for each bug file; done; add xxx, yyy, zzz to code corpuses; done; add function compute_CodeCorpus_norm; done; compute Norm for each code file; done; (2019-9-24)
312.Let’s neglect g(#terms) first; compute the intersection of q and d; done; compute rVSM score; done; add attribute rVSMScore to class Corpus; done; compute each rVSM score for CodeCorpus; done; (2019-9-24)
313.separate folder path; done; let’s try to change path and test the result; done; (2019-9-24)
314.Use numpy to compute log; give up; (2019-9-25)
315.Let’s take the length of code length into account; change function name to compute_rVSMScore; done; add DocumentLength attribute to code corpus; done; the length of document can be seen in CodeFiles[i], compute them; done; compute xmax and xmin; done; add attribute Nx to corpus; done; compute Nx for each code file; move Min and Max to code files line; done; delete DocumentLength attribute; done; (2019-9-25)
316.compute N(x) for each code document; done; how to calculate exp in python; math.exp( x ); done; add g(#term) to compute rVSM score; done; python how to rank list; 在这里，可以看出sort()是没有返回值的，它会改变原有的列表，而sorted()需要用一个变量进行接收，它并不会修改原有的列表; try to sort according to rVSMScore; done; (2019-9-25)
317.Output file name of code file; done; let us test this on the original path; test 100040.txt; WELL DONE; test 100095.txt; WELL DONE; (2019-9-25)
318.Read how they deal with similar bugs; #term is normalized; separate BugCorpus and CodeCorpus; done; delete Nx, Norm and rVSMScore in class BugCorpus; done; change path to test result; done; (2019-9-26)
319.Find max and min of rVSMScore; python get last element in a list; give up; normalize rVSMScore; done; (2019-9-26)
320.----------------------------------------------------------------------------------------------------------------------
321.Compute similarity between bugs; add similarity attribute to class BugCorpus; done; write function compute_Similarity; done; strange, while the three answers are not the same? Add Norm to class BugCorpus; done; FileName is right; Norm is right; the problem is in compute_intersection; change compute_intersection to compute_BugCode_intersection; done; change path to test the result; correct; add function compute_BugBug_intersection; done; test function compute_Similarity; done; (2019-9-26)
322.Check tomorrow class; no class tomorrow; done; (2019-9-27)
323.before that, let us separate variable BugNumber; done; Allocate similarity value to each bug corpus; done; delete extra files in folder BugLocatorGY; done; (2019-9-27)
324.Add attribute RelevantBugs to class CodeCorpuses; done; read FixLink.txt into 2D array; done; which RelevantBugs should store, file name or file path? I think we should store complete file path; (2019-9-27)
325.python how to find class list index; give up; create a list to contain code file name; done; change path to test result; append bug file name to each code corpus; test the result; strange, the same bug file name repeat many times; it is because our cell has run many times; solved; (2019-9-27)
326.Read slides of big data 1; (2019-9-27)
327.个人理解守护进程和普通进程区别是指：将后台程序变成一种服务，比如说，用命令行输入启动程序，如果不是守护进程的话，一旦命令行窗口关闭，程序就终止了；而如果启动守护进程，则退出命令行窗口之后，服务一直处于运行状态。(2019-9-27)
328.libevent是一个异步事件处理软件库，以BSD许可证发布。libevent提供了一组应用程序编程接口（API），让程序员可以设置某些事件发生时所运行的函数，也就是说，libevent可以用来取代网络服务器所使用的事件循环检查框架。由于可以省去对网络的处理，且拥有不错的性能，有些软件使用libevent作为网络底层的库，如：memcached、Tor。(2019-9-27)
329.最久未使用算法（LRU）：最久没有访问的内容作为替换对象; (2019-9-27)
330.----------------------------------------------------------------------------------------------------------------------
331.Set bug similarity with its own to -1; done; add function compute_SimiScore; add '\\BugCorpus\\' to file path; done; create bug file name list so as to find index; done; set BugFileNames as a parameter; done; change variable name BugNumber to LocateBugNumber; done; decide if counter is zero; done; add class SimiScore to class CodeCorpus; done; compute SimiScore for all code corpuses; done; normalize them done; (2019-9-27)
332.Add FinalScore attribute to class CodeCorpus; done; compute final score for all code corpuses; done; rank them; done; WELL DONE; final score is really better than two before; done; (2019-9-27)
333.Check format of output; place compute rVSMScore, SimiScore and FinalScore in the same cell; delete rVSM sort; the result is almost the same, we can say that we are nearly correct; let’s check point of each bug for 10040.txt; not as good as the result shows; hold; (2019-9-28)
334.We hope to output rank and score of bug-link-code; done; add attribute RelevantCodes to class BugCorpus; done; change CodeFileNumber to CodeNumber; done; change ThisBugFileName to ThisBugName; done; change ThisCodeFileName to ThisCodeName; done; (2019-9-28)
335.Add rank attribute to class CodeCorpuses; done; allocate the result rank to each code corpus; done; since we need rank attribute, so we use sort instead of sorted; done; combine all cells into a single one; store the result in a 2D array; WELL DONE; FINALLY; (2019-9-28)
336.Amazon DynamoDB is a fully managed proprietary NoSQL database service that supports key-value and document data structures and is offered by Amazon.com as part of the Amazon Web Services portfolio. (2019-9-28)
337.Write slides of Zhou [2012]; hold; (2019-9-29)
338.Deal with computing intelligence; figure out the construction of 4/3; done; figure out the construction of 1/2; done; (2019-9-29)
339.在数学中，线性规划（Linear Programming，简称LP）特指目标函数和约束条件皆为线性的最优化问题。线性规划是最优化问题中的一个重要领域。在作业研究中所面临的许多实际问题都可以用线性规划来处理，特别是某些特殊情况，例如：网络流、多商品流量等问题，都被认为非常重要。目前已有大量针对线性规划算法的研究。很多最优化问题算法都可以分解为线性规划子问题，然后逐一求解。在线性规划的历史发展过程中所衍伸出的诸多概念，建立了最优化理论的核心思维，例如“对偶”、“分解”、“凸集”的重要性及其一般化等。在微观经济学和商业管理领域中，线性规划亦被大量应用于例如降低生产过程的成本等手段，最终提升产值与营收。乔治·丹齐格被认为是线性规划之父。(2019-9-29)
340.----------------------------------------------------------------------------------------------------------------------
341.Download homework of computing intelligence; (2019-9-29)
342.Download attachment of scholarship; read it again; submit; done; (2019-10-8)
343.申请理由: 
344.今年，我通过全国硕士研究生统一招生考试，荣幸成为了一名华中科技大学计算机学院的硕士研究生。在学习方面，我上课认真听讲，按时完成作业和报告，不懂的地方积极向老师和同学请教。在科研方面，我每周通过邮件向导师汇报进展，定期和导师面对面讨论科研工作，和实验室的同学融洽相处。在生活方面，我保持规律的生活作息，早睡早起；每天坚持体育锻炼，为科研工作打下一个坚实的身体基础。综上，我希望能够获得研究生学业奖学金，作为我读研期间的生活保障。在今后，我会一以贯之地努力奋斗，成为一名优秀的华中科技大学研究生。
345.Finish slides of Zhou [2012]; done; send request to Professor; (2019-10-8)
346.Download slides of chapter 2, distributed system; done; (2019-10-8)
347.传奇私服; (2019-10-8)
348.当处理复杂数据感到头大时，可以自己构造一些简单的数据。在简单的数据上验证了结构后，再去处理复杂数据。这类似于线性代数中的同构思想。(2019-10-8)
349.Java远程方法调用，即Java RMI（Java Remote Method Invocation）是Java编程语言里，一种用于实现远程过程调用的应用程序编程接口。它使客户机上运行的程序可以调用远程服务器上的对象。远程方法调用特性使Java编程人员能够在网络环境中分布操作。RMI全部的宗旨就是尽可能简化远程接口对象的使用。
350.----------------------------------------------------------------------------------------------------------------------
351.套接字（socket）是一个抽象层，应用程序可以通过它发送或接收数据，可对其进行像对文件一样的打开、读写和关闭等操作。套接字允许应用程序将I/O插入到网络中，并与网络中的其他应用程序进行通信。网络套接字是IP地址与端口的组合。
352.消息传递接口（英语：Message Passing Interface，缩写MPI）是一个并行计算的应用程序接口（API），常在超级计算机、计算机集群等非共享内存环境程序设计。
353.分布式共享存储器（英语：Distributed shared memory，简称DSM）, 也被视为一种分散的全局地址空间 (Distributed Global Address Space), 属于计算机科学的一种机制，可以透过硬件或软件来实现。分布式共享存储器主要使用在集群计算机中，集群计算机中的每一个网络结点(node)都有非共享的存储器空间与共享的存储器空间。该共享存储器的位置空间(address space)在所有结点是一致的。简单说，同一时间下在结点A读取0x00001234会和结点B读取0x00001234得到一样的值。
354.Start read Ming Li’s work; Learning Unified Features from Natural and Programming Languages for Locating Buggy Source Code; done; (2019-10-9)
355.Review abstract; introduction 1; introduction 2, locate source file; introduction 3, previous work; lose information by ignoring program structure; done; (2019-10-9)
356.Review related work; most models ignore structural information; apply deep learning to software engineering; done; (2019-10-9)
357.Download font pack of adobe reader; done; print scholarship file; done; (2019-10-9)
358.Google文件系统（英语：Google File System，缩写为GFS或GoogleFS），一种专有分布式文件系统，由Google公司开发，运行于Linux平台上。尽管Google在2003年公布了该系统的一些技术细节，但Google并没有将该系统的软件部分作为开源软件发布。(2019-10-9)
359.Now we have 3 problems; matrix; big database; distributed systems; (2019-10-10)
360.----------------------------------------------------------------------------------------------------------------------
361.猎人学校; (2019-10-10)
362.线性代数中的自然基是指由某一维为1其余维都为0的向量组成的一组基，基里的向量均线性无关; (2019-10-10)
363.download octave; done; install octave; done; calculate inverse and multiply; done; (2019-10-10)
364.Write progress report; send; done; (2019-10-11)
365.Order banana online; order mi pad online; done; (2019-10-11)
366.Deal with the read-only problem in one-drive; done; (2019-10-11)
367.Continue read slides of big data; give up; (2019-10-11)
368.Modify research experience; done; (2019-10-11)
369.Order print paper online; chat with Liu about print papers; done; (2019-10-12)
370.----------------------------------------------------------------------------------------------------------------------
371.Get mi pad; done; set up mi pad; done; set password for mi pad; done; (2019-10-12)
372.线性规划 is linear programming; IP is integer programming; done; (2019-10-14)
373.Search google for job scheduling problems using Chinese; not found; try to search related work; find it; done; (2019-10-14)
374.A greedy heuristic for the set-covering problem； Mathematics of Operations Research; done; (2019-10-14)
375.Submit computing intelligence assignment to Brother Yang; done; (2019-10-14)
376.微信913+支付宝3395+建行2818+中行2472; done; (2019-10-15)
377.Check where is Jordan normal form in Peking’s lecture; it is in episode 114; it seems that we cannot find relevant content in Peking’s lecture; done; (2019-10-15)
378.Yanhao Yang accept my computing intelligence homework; (2019-10-15)
379.Add dimension to word norm; (2019-10-15)
380.----------------------------------------------------------------------------------------------------------------------
381.ACID，是指数据库管理系统（DBMS）在写入或更新资料的过程中，为保证事务（transaction）是正确可靠的，所必须具备的四个特性：原子性（atomicity，或称不可分割性）、一致性（consistency）、隔离性（isolation，又称独立性）、持久性（durability）。
382.二阶段提交（英语：Two-phase Commit）是指在计算机网络以及数据库领域内，为了使基于分布式系统架构下的所有节点在进行事务提交时保持一致性而设计的一种算法。通常，二阶段提交也被称为是一种协议（Protocol）。在分布式系统中，每个节点虽然可以知晓自己的操作时成功或者失败，却无法知道其他节点的操作的成功或失败。当一个事务跨越多个节点时，为了保持事务的ACID特性，需要引入一个作为协调者的组件来统一掌控所有节点（称作参与者）的操作结果并最终指示这些节点是否要把操作结果进行真正的提交（比如将更新后的数据写入磁盘等等）。因此，二阶段提交的算法思路可以概括为： 参与者将操作成败通知协调者，再由协调者根据所有参与者的反馈情报决定各参与者是否要提交操作还是中止操作。
383.与两阶段提交不同的是，三阶段提交是“非阻塞”协议。三阶段提交在两阶段提交的第一阶段与第二阶段之间插入了一个准备阶段，使得原先在两阶段提交中，参与者在投票之后，由于协调者发生崩溃或错误，而导致参与者处于无法知晓是否提交或者中止的“不确定状态”所产生的可能相当长的延时的问题[1]得以解决。 举例来说，假设有一个决策小组由一个主持人负责与多位组员以电话联络方式协调是否通过一个提案，以两阶段提交来说，主持人收到一个提案请求，打电话跟每个组员询问是否通过并统计回复，然后将最后决定打电话通知各组员。要是主持人在跟第一位组员通完电话后失忆，而第一位组员在得知结果并执行后老人痴呆，那么即使重新选出主持人，也没人知道最后的提案决定是什么，也许是通过，也许是驳回，不管大家选择哪一种决定，都有可能与第一位组员已执行过的真实决定不一致，老板就会不开心认为决策小组沟通有问题而解雇。三阶段提交即是引入了另一个步骤，主持人打电话跟组员通知请准备通过提案，以避免没人知道真实决定而造成决定不一致的失业危机。为什么能够解决二阶段提交的问题呢？回到刚刚提到的状况，在主持人通知完第一位组员请准备通过后两人意外失忆，即使没人知道全体在第一阶段的决定为何，全体决策组员仍可以重新协调过程或直接否决，不会有不一致决定而失业。那么当主持人通知完全体组员请准备通过并得到大家的再次确定后进入第三阶段，当主持人通知第一位组员请通过提案后两人意外失忆，这时候其他组员再重新选出主持人后，仍可以知道目前至少是处于准备通过提案阶段，表示第一阶段大家都已经决定要通过了，此时便可以直接通过。
384.数据加密标准（英语：Data Encryption Standard，缩写为 DES）是一种对称密钥加密块密码算法，1976年被美国联邦政府的国家标准局确定为联邦资料处理标准（FIPS），随后在国际上广泛流传开来。它基于使用56位密钥的对称算法。这个算法因为包含一些机密设计元素，相对短的密钥长度以及怀疑内含美国国家安全局（NSA）的后门而在开始时有争议，DES因此受到了强烈的学院派式的审查，并以此推动了现代的块密码及其密码分析的发展。
385.哪有现在看动漫的; (2019-10-22)
386.Deal with Homework 3 of computing intelligence; download Homework 3; done; Task Scheduling Problem; T对应的最优解X中包含mn个变量，确定mn个变量需要mn个
387.线性无关等式; find related slides of Homework 3; it is file 10.pdf; done; (2019-10-25)
388.Search google for Non-Uniform Job Scheduling; Theory and Practice in Parallel Job Scheduling is not the answer; search google for Non-Uniform Job Scheduling approximate algorithm; (2019-10-26)
389.Search articles that cite this paper; read an article discussing bin packing; give up; search google for LP relaxation; 
390.----------------------------------------------------------------------------------------------------------------------
391.find a useful link http://timroughgarden.org/w16/l/; lecture 1, introduction to maximum flow; lecture 2, augmenting path for maximum flow; lecture 3, push-relabel for maximum flow; lecture 4, application of maximum flow; lecture 5, matching; lecture 6, generalizations; lecture 7, introduction to linear programming; lecture 8, Linear Programming Duality part 1; lecture 9, Linear Programming Duality part 2; lecture 10, the minimax theorem; lecture 11, online learning; lecture 12, applications; lecture 13, online scheduling; lecture 14, Online Bipartite Matching; lecture 15, introduction to approximation algorithm; lecture 16, TSP; (2019-10-26)
392.Search google for linear programming relaxation; done; (2019-10-29)
393.数据库连接池负责分配、管理和释放数据库连接，它允许应用程序重复使用一个现有的数据库连接，而不是再重新建立一个；释放空闲时间超过最大空闲时间的数据库连接来避免因为没有释放数据库连接而引起的数据库连接遗漏。这项技术能明显提高对数据库操作的性能。
394.若矩阵A=[a1，a2，…，an]∈Cm×n为复矩阵，则其列向量的所有线性组合的集合构成一个子空间，称为矩阵A的列空间(column space)或列张成(column span)，用符号CoI(A)表示。
395.861+3486+2472+3563
396.理发; 洗澡; (2019-11-1)
397.WPS换行不顶格; give up; ask where can I buy examination papers of previous year; they send we chat account to us; add we-chat friends to list; 东九对面，绅宝驾校旁，游泳馆快印; done; ask whether they have examination papers of previous year; they said no, ah~~~; (2019-11-2)
398.Opening ceremony of zhiyuan; download slides of distributed system; chapter 4 is out of data, ask for help in QQ group; download it; done; (2019-11-2)
399.Deal with the problem of printing phrases; Too few arguments to function print_sentence(); done; (2019-11-2)
400.----------------------------------------------------------------------------------------------------------------------
401.共享和CS也有共通之处, 说成是可重用性更为贴切; (2019-11-2)
402.When confronting complex mathematics equations in programming, we can separate sigma to increase readability of our code. (2019-11-2)
403.It’s not who has the best algorithm that wins. It’s who has the most data; (2019-4-23)
404.Apache Hadoop是一款支持数据密集型分布式应用程序并以Apache 2.0许可协议发布的开源软件框架。它支持在商品硬件构建的大型集群上运行的应用程序。Hadoop是根据谷歌公司发表的MapReduce和Google文件系统的论文自行实现而成。所有的Hadoop模块都有一个基本假设，即硬件故障是常见情况，应该由框架自动处理。(2019-11-3)
405.Apache ZooKeeper是Apache软件基金会的一个软件项目，它为大型分布式计算提供开源的分布式配置服务、同步服务和命名注册。ZooKeeper曾经是Hadoop的一个子项目，但现在是一个独立的顶级项目。(2019-11-3)
406.try to search github for approximation algorithm; it is hard to find; done; (2019-11-5)
407.Read papers of computing intelligence; download a paper discussing Generalized assignment problem; (2019-11-5)
408.There are a number of agents and a number of tasks. Any agent can be assigned to perform any task, incurring some cost and profit that may vary depending on the agent-task assignment. Moreover, each agent has a budget and the sum of the costs of tasks assigned to it cannot exceed this budget. It is required to find an assignment in which all agents do not exceed their budget and total profit of the assignment is maximized. (2019-11-5)
409.Search google for which problem can be used approximation algorithm; read a general introduction of GAP; pij, profit of item j if assigned to knapsack I; wij, weight of item j if assigned to knapsack I; ci, capacity of knapsack I; (2019-11-5)
410.----------------------------------------------------------------------------------------------------------------------
411.Ask Yong Yuan the examination data of matrix theory; he said it is at the end of this semester; done; (2019-11-5)
412.Ask Liu whether 文本的相似度比对 is pattern recognition or not; yes, we can; (2019-11-5)
413.We can use review to check what we learn; (2019-11-7)
414.LISP名称源自列表处理（LISt Processing）的英语缩写，由来自麻省理工学院的人工智能研究先驱约翰·麦卡锡（John McCarthy）在1958年基于λ演算所创造，采用抽象数据列表与递归作符号演算来衍生人工智能 。(2019-11-7)
415.指令式编程（英语：Imperative programming）；是一种描述电脑所需作出的行为的编程典范。几乎所有电脑的硬件工作都是指令式的；几乎所有电脑的硬件都是设计来运行机器代码，使用指令式的风格来写的。较高端的指令式编程语言使用变量和更复杂的语句，但仍依从相同的典范。菜谱和行动清单，虽非计算机程序，但与指令式编程有相似的风格：每步都是指令，有形的世界控制情况。因为指令式编程的基础观念，不但概念上比较熟悉，而且较容易具体表现于硬件，所以大部分的编程语言都是指令式的。(2019-11-7)
416.grep是一个最初用于Unix操作系统的命令行工具。在给出文件列表或标准输入后，grep会对匹配一个或多个正则表达式的文本进行搜索，并只输出匹配（或者不匹配）的行或文本。(2019-11-7)
417.Let’s check what is [4]; download it; (2019-11-8)
418.Let’s check what is [5]; 
419.孔子有两个学生，一个做好事收了钱，一个做好事没收钱。孔子表扬了做好事收钱的学生，批评了做好事不收钱的学生。孔子的理由是，如果天下人都认为做好事不能收钱，那么天底下就没几个人愿意做好事了，费事费力不说，还会蒙受经济的损失。反过来，如果天下人都认为做好事可以收钱，那么大家都会争先恐后地做好事。这种风气不仅可以让更多有需要的人得到帮助，还可以给做好事的人带来经济上的补偿，你好我好大家好。
420.----------------------------------------------------------------------------------------------------------------------
421.2472+3538+3471+833; buy a cell phone for haiwangxing; (2019-11-12)
422.Against will as friends circle; (2019-11-13)
423.Write sketch of each chapter; done; (2019-11-14)
424.Download slides of distributed system; done; (2019-11-14)
425.Against will to see ipad; (2019-11-16)
426.Against will to see QQ; (2019-11-16)
427.Against will to see QQ; (2019-11-16)
428.accept apple and pear; done; (2019-11-19)
429.Check whether zhangqizhifu has already repaired; not yet; order tissue on Jingdong; carry them back; done; download receipt; send receipt to Xuelian; done; (2019-11-21)
430.----------------------------------------------------------------------------------------------------------------------
431.Send code and PPT to Zhixin Xu; order clock online; done; order a box of apple online; done; (2019-11-21)
432.Send group name and information to Prof. Liu; done; (2019-11-21)
433.Download slides of pattern recognition; done; (2019-11-21)
434.先用python写验证可行性, 验证了之后再用C++写来提高执行效率; (2019-11-16)
435.Read paper regarding approximate algorithm; it file name is st; done; (2019-11-22)
436.Against will to see weibo; (2019-11-22)
437.Start to work afternoon; (2019-11-22)
438.Against will to see tmall; (2019-11-22)
439.What is xij; xij=0/1, whether job j is on machine i; (2019-11-22)
440.----------------------------------------------------------------------------------------------------------------------
441.What is LP(t)? integer solutions to the following linear program; (2019-11-22)
442.Send weekly report to Prof. He; (2019-11-22)
443.Against will; (2019-11-23)
444.AG; (2019-11-24)
445.AG; (2019-11-24)
446.Order printer paper online; done; ask Zhixin to take paper back; wait for bill; send bill to Xuelian; (2019-11-22)
447.I think we can translate the content of paper and use our notes to deal with pattern recognition report; let us start to translate; (2019-11-24)
448.Download slides of AI course; done; (2019-11-26)
449.Read slides of AI chapter 1; done; (2019-11-26)
450.----------------------------------------------------------------------------------------------------------------------
451.Read slides of AI chapter 2; done; (2019-11-28)
452.I hope to deal with graph search in report of AI; (2019-11-26)
453.Upload slides of contributed system to Baidu cloud; done; (2019-11-28)
454.Read slides of AI chapter 3; 
455.Download report topic of AI; already downloaded; what is open table and close table? OPEN表：用于存放刚生成的节点；CLOSE表：用于存放将要扩展或已扩展的节点；(2019-11-28)
456.Review 坐标，过渡矩阵，基；done; (2019-11-29)
457.简称, added; (2019-11-30)
458.Show sample bug vector and code vector; done; (2019-11-30)
459.Check our code how to compute inner product; done; (2019-12-1)
460.----------------------------------------------------------------------------------------------------------------------
461.Add index to each equations; (2019-12-1)
462.Ask about examination paper of previous year; buy them; (2019-12-1)
463.What is Hermite matrix; ai,j = 共轭(aj,i); (2019-12-3)
464.What is 幂等矩阵？AA=A; (2019-12-3)
465.Search google for 广义逆; I think we need to review as sequence; (2019-12-3)
466.准对角矩阵; done; (2019-12-3)
467.Order hair dryer online; order fruit online; (2019-12-4)
468.Drawing diagram or schematic by hand helps us understand difficult theory, since we are better at imagery thinking than abstract thinking; (2019-12-5)
469.Review what is linear map on the textbook; done; (2019-12-5)
470.----------------------------------------------------------------------------------------------------------------------
471.Finish content of Chapter 6; done; (2019-12-5)
472.Send progress report of last week; (2019-12-5)
473.Download slides of distributed system of chapter 8; done; (2019-12-5)
474.矩阵A的谱半径等于矩阵A的特征值绝对值的最大值。(2019-12-6)
475.求矩阵特征值; done; (2019-12-6)
476.V按列分块为(v1, v2, ..., vn), 它的n个列是对应于特征值的标准正交的特征向量; (2019-12-6)
477.求逆矩阵的方法; 伴随矩阵法; (2019-12-6)
478.在数论中，一个双线性映射是由两个向量空间上的元素，生成第三个向量空间上一个元素之函数，并且该函数对每个参数都是线性的。例如矩阵乘法就是一个例子。(2019-12-7)
479.有几个线性无关的特征向量就是有几个几何重数; (2019-12-7)
480.----------------------------------------------------------------------------------------------------------------------
481.代数重数确定Jordan矩阵的阶数, 几何重数确定Jordan矩阵中Jordan块数; (2019-12-7)
482.若代数重数=几何重数, 则Jordan矩阵为对角矩阵; (2019-12-7)
483.The question is, how can we get vector beta? Beta starts from 2; done; (2019-12-7)
484.What is nj? 
485.方法一（计算可逆矩阵）太过复杂，一般不采用; (2019-12-7)
486.Why the rank of two vectors is 1? Outer Product generates the matrix whose first row is u1(v1,v2,..,vn) and the ith row is ui(v1,v2,..,vn). So the rows are the vector (v1,v2,..,vn) multiplied by scalars. So this itself is the basis.Hence dimension is 1. (2019-12-8)
487.Why idempotent matrix’s eigenvalue is 0 and 1? Ax=lambda*x; AAx=lambda^2*x; (2019-12-8)
488.矩阵函数的特征多项式和最小多项式是什么? (2019-12-8)
489.What is the strange derivative? It is function of Jordan block; (2019-12-8)
490.----------------------------------------------------------------------------------------------------------------------
491.怎样求线性变换的不变子空间? 没有直接的办法，取一个坐标轴，再求它的正交补空间; (2019-12-8)
492.Change format of equation in report to Math type; done; (2019-12-8)
493.什么是等代价搜索？
494.We should also review the whole paper to make the PPT; done; (2019-12-10)
495.看一看证明题该怎么搞; (2019-12-12)
496.Test minimal polynomial on octave; 最小多项式是化零多项式中次数最低，首项系数为1的多项式; done; (2019-12-12)
497.线性变换的矩阵能够怎么用; T(x)=Ax; (2019-12-12)
498.Send weekly report to Prof. He; done; (2019-12-12)
499.Set outline of report; done; (2019-12-12)
500.----------------------------------------------------------------------------------------------------------------------
